### PR TITLE
Get parameters for Plaid link token based on platform

### PR DESCRIPTION
**Pull request by marcaaron**
~~cc @Dal-Papa unsure if we have yet added the necessary `redirect_uri` and `android_package_name` fields to the production instance of Plaid, but I think these native versions will not work properly until we do.~~ This is done now

### Details
Passes the `redirect_uri` and `android_name` fields to `Plaid_GetLinkToken` command

### Fixed Issues
IDK if there is a specific issue. It's more of a follow up to: https://github.com/Expensify/App/pull/6362

### Tests

I tested this using the steps in the linked PR above. However, also saw that the OAuth stuff is not testable with the Plaid sandbox. 

I think we won't know for sure unless we test against production, but can simulate this by modifying our local `Web-Secure to use the Plaid dev app Development secret and connecting with BofA.

### QA Steps

Internal / See Comments below

### Tested On

- [ ] Web
- [ ] Mobile Web
- [ ] Desktop
- [x] iOS
- [x] Android

### CODE REVIEW

**Comment by marcaaron** [comment]
@nickmurray47 adding you here since I think these changes might intersect with the work you are doing for https://github.com/Expensify/Expensify/issues/160148

**Comment by marcaaron** [comment]
I think we probably want to hold this PR on confirmation that the android package name and redirect uri for iOS are registered with the production version of our Plaid instance.

File: src/libs/getPlaidLinkTokenParameters/index.ios.js
Comment (nickmurray47) [inline src/libs/getPlaidLinkTokenParameters/index.ios.js:3]: should we add a staging URL here or assume we're just going to test on prod?

**Review by nickmurray47 — COMMENTED** [review]
Looks good to me! On this: 
> tested this using the steps in the linked PR above. However, also saw that the OAuth stuff isn't really testable with the Plaid sandbox

Does this mean the Platypus OAuth sandbox credentials are not sufficient? What do we need to test for in this PR (or are we just gonna test the whole flow together)?

**Comment by marcaaron** [comment]
>Does this mean the Platypus OAuth sandbox credentials are not sufficient? What do we need to test for in this PR (or are we just gonna test the whole flow together)?

Good question. I'm actually not sure yet, but the docs had this to say:



I'm guessing the "sandbox-only" institution they're referring to is Platypus OAuth, but really unsure if our "dev" instance that everyone can access is set up for OAuth testing or not.

I think what we need to test is:

- The production instance of Plaid a.k.a. we need to hit Web-Secure prod to get the right token (we can do this locally by passing in the prod web-secure endpoint)
- Once merged, verify we can go through one of the approved OAuth bank flows on staging (while switched to the prod token). Looking at the most recent pic that @flodnv shared [here](https://expensify.slack.com/archives/C02NQJMCQ6M/p1637690194015200?thread_ts=1637676798.009100&cid=C02NQJMCQ6M) it seems like we can only test a Capital One connection. I'm confused about why some of these other banks are "Processing" - but guessing it has to do with some missing required information.

File: src/libs/getPlaidLinkTokenParameters/index.ios.js
Comment (marcaaron) [reply src/libs/getPlaidLinkTokenParameters/index.ios.js:3]: Ah hmm that's a good question. This url will be whatever we set it to here...

https://github.com/Expensify/App/blob/e24b86b5217ef885f6da0eb1b17bbcfec143d13d/src/CONFIG.js#L10
https://github.com/Expensify/App/blob/e24b86b5217ef885f6da0eb1b17bbcfec143d13d/.env.staging#L1
https://github.com/Expensify/App/blob/e24b86b5217ef885f6da0eb1b17bbcfec143d13d/.env.production#L1

But my understanding is that it will only be used to redirect back to the app and not actually navigate us anywhere. And in that case the iOS app has universal links set up for both staging and production - see here:

https://github.com/Expensify/App/blob/f31be6f12404bb8ed7aecb0115d422e720a0a297/ios/NewExpensify/Chat.entitlements#L9-L13

Which means that a link to either endpoint will always be handled as a universal link by the app. So basically, my guess is that it doesn't matter at all. LMK if that makes sense.

**Review by nickmurray47 — DISMISSED** [review]
this all sounds good and lgtm

**Comment by marcaaron** [comment]
Chatting with @Dal-Papa now about testing for Mobile-Expensify, but I think we can use a similar testing strategy for this PR to verify that mobile is working. Which is:

1. Update the Web-Secure local to use our Development secret key
2. Use ngrok to point the local native apps at the local web-secure
3. Go through the VBA flow and select BofA (the only bank set up for OAuth in our Plaid dev app atm)
4. See if we can complete the flow in iOS and Android
5. Merge this PR
6. Test again on staging but with the www-secure not staging-secure

I don't have a BofA account to test this with otherwise I'd do it myself. Going to make this internal QA in any case.

**Review by nickmurray47 — DISMISSED** [review]

**Comment by marcaaron** [comment]
Ok this is ready to be merged - but I think we are still stuck since we need actual accounts set up with participating banks to test and are only able to test BofA with the Plaid dev app anyway. Thinking more on this... really we should test them all to make sure they work before the change is deployed so we might as well use the **Plaid production app Development secret**. 

IMO we shouldn't skip this step (and should do it for all the participating banks) because we have no idea if those connections will actually work unless we test them first.

Two options I can think of:

1. Find engineer volunteers with the various banks to test this stuff and share the **Plaid production app Development secret** with them and have them hardcode it in their local `Web-Secure` then test the flows on native platforms.

2. Just merge this and hope everything works on staging - but if it _doesn't_ work we'll need an engineer with access to one of those banks to test the flow anyway and we will need to go back to option `1` anyway.

**Comment by marcaaron** [comment]
Discussed a 3rd option with @nickmurray47 which is to add a beta and allow non engineers to also test the flow. But if we do this idea we still need to solve for how to "switch" to the Development secret for test purposes under the beta.

Ultimately, I'm not sure it's worth building out that kind of functionality yet - but one benefit is that it opens up more testing options (i.e. non engineers could test) and enable us to switch on the beta once testing is complete (even if that happens after 11/30).

**Review by marcochavezf — DISMISSED** [review]
The changes look good to me (I was catching up on the OAuth migration to understand this a little bit more) 👍🏽

File: src/CONST.js
Comment (Dal-Papa) [inline src/CONST.js:7]: Can you explain this to me please ? I'm curious why it's not a regular const.

**Review by Dal-Papa — DISMISSED** [review]

File: src/CONST.js
Comment (marcaaron) [reply src/CONST.js:7]: Actually was trying to do something [fancy](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Functions/get) here but it broke everything on web 😄

Comment (marcaaron) [reply src/CONST.js:7]: Basically a getter let's you access `this.someObjectProperty` from within the getter. But it does not work the other way around 🤦

**Review by Dal-Papa — DISMISSED** [review]
Looks even better to me now that I get it 😆

**Comment by VictoriaExpensify** [comment]
@marcaaron once this is implemented, would this resolve the Plaid issue that Chase customers were having in [this GH](https://github.com/Expensify/Expensify/issues/145418)?

**Comment by marcaaron** [comment]
@VictoriaExpensify Looks unrelated to me as OAuth should not be forced for any users yet.

**Review by marcochavezf — APPROVED** [review]

**Comment by botify** [comment]
@marcochavezf looks like this was merged without passing tests. Please add a note explaining why this was done and remove the `Emergency` label if this is not an emergency.

**Comment by marcaaron** [comment]
Gonna check this off the deploy list. There doesn't seem to be a clear way to test this beyond regressions for the VBA flow. It's basically Production QA.

**Comment by marcochavezf** [comment]
There is no emergency here, so I'm removing the label. Seems the checklist was not completed

**Pull request merged**
